generator/src/lib.rs

```diff
@@ -2032,8 +2020,18 @@ pub fn derive_setters(
         quote!()
     };
 
-    // If the struct extends something we need to implement the trait.
-    let impl_extend_trait = root_structs.iter().map(|extends| {
+    // If the struct extends something we need to implement the traits.
+    let extends_structs = _struct
+        .extends
+        .as_ref()
+        .map(|extends| {
+            extends
+                .split(',')
+                .map(|extends| format_ident!("Extends{}", name_to_tokens(extends)))
```

Comment:
I wonder if at some point we should check if the `extends` name exists within `extends_structs`, if we ever perform more filtering on this `HashSet` in the future (ie. ignoring certain structs).

Reply:
YAGNI ?

Reply:
We'll run into it whenever.